charmhelpers/contrib/network/ovs/__init__.py

```diff
@@ -77,30 +77,22 @@ def del_bridge_port(name, port):
     subprocess.check_call(["ip", "link", "set", port, "promisc", "off"])
 
 
-def add_ovsbridge_linuxbridge(name, bridge):
+def add_ovsbridge_linuxbridge(name, bridge, promisc=False):
     ''' Add linux bridge to the named openvswitch bridge
     :param name: Name of ovs bridge to be added to Linux bridge
     :param bridge: Name of Linux bridge to be added to ovs bridge
     :returns: True if veth is added between ovs bridge and linux bridge,
     False otherwise'''
-    try:
-        import netifaces
-    except ImportError:
-        if six.PY2:
-            apt_install('python-netifaces', fatal=True)
-        else:
-            apt_install('python3-netifaces', fatal=True)
-        import netifaces
 
     ovsbridge_port = "veth-" + name
     linuxbridge_port = "veth-" + bridge
     log('Adding linuxbridge {} to ovsbridge {}'.format(bridge, name),
         level=INFO)
-    interfaces = netifaces.interfaces()
-    for interface in interfaces:
-        if interface == ovsbridge_port or interface == linuxbridge_port:
```

Comment:
This seems to remove the check for `linuxbridge_port` already existing.  Is that intended?